packages/react-server-test-pages/pages/navigation/forward.js

```diff
@@ -0,0 +1,50 @@
+import {ReactServerAgent} from "react-server";
+
+export default class ForwardPage {
+	handleRoute() {
+		const request = this.getRequest();
+		let params = request.getQuery();
+		if (params) {
+			params = params.value;
+		} else {
+			params = 0;
+		}
+
+		//fetch some data
+		this.data = ReactServerAgent.get('/data/delay?ms=1000&val='+params);
+
+		return this.data.then((res) => {
+			//then depending on said data, forward to one of two pages, and pass along the data we pre-fetched
+			if (res.body % 2 === 0) {
+				if (typeof window !== 'undefined') { //would be nice if this is `process.env.isServer`
+					require.ensure(["./forwardEven"], () => {
```

Comment:
I can't quite tell because babelification messes with line numbers, but it _looks_ like the problem is that you're not returning the `require.ensure` result.